File: literalura-frontend/src/services/autorService.js
import api from './api'

const autorService = {
  // Listar todos los autores
  getAll: async () => {
    const response = await api.get('/autores')
    return response.data
  },

  // Obtener autor por ID
  getById: async (id) => {
    const response = await api.get(`/autores/${id}`)
    return response.data
  },

  // Buscar autor por nombre
  buscarPorNombre: async (nombre) => {
    const response = await api.get('/autores/buscar', {
      params: { nombre }
    })
    return response.data
  },

  // Autores vivos en un año
  getVivosEnAnio: async (anio) => {
    const response = await api.get('/autores/vivos', {
      params: { anio }
    })
    return response.data
  },

  // Autores nacidos entre años
  getNacidosEntre: async (desde, hasta) => {
    const response = await api.get('/autores/nacidos', {
      params: { desde, hasta }
    })
    return response.data
  },

  // Autores fallecidos entre años
  getFallecidosEntre: async (desde, hasta) => {
    const response = await api.get('/autores/fallecidos', {
      params: { desde, hasta }
    })
    return response.data
  }
}

export default autorService

File: backend/src/main/java/com/literalura/challenge_java/service/AutorService.java
package com.literalura.challenge_java.service;

import com.literalura.challenge_java.dto.AutorDTO;
import com.literalura.challenge_java.entity.Autor;
import com.literalura.challenge_java.mapper.AutorMapper;
import com.literalura.challenge_java.repository.AutorRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@Transactional(readOnly = true)
public class AutorService {

    private final AutorRepository autorRepository;
    private final AutorMapper autorMapper;

    public AutorService(AutorRepository autorRepository, AutorMapper autorMapper) {
        this.autorRepository = autorRepository;
        this.autorMapper = autorMapper;
    }

    /**
     * Lista todos los autores registrados
     */
    public List<AutorDTO> listarTodos() {
        List<Autor> autores = autorRepository.findAll();
        return autorMapper.toDTOList(autores);
    }

    /**
     * Obtiene un autor por su ID
     */
    public AutorDTO obtenerPorId(Long id) {
        Autor autor = autorRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Autor no encontrado con ID: " + id));
        return autorMapper.toDTO(autor);
    }

    /**
     * Busca autores por nombre
     */
    public List<AutorDTO> buscarPorNombre(String nombre) {
        List<Autor> autores = autorRepository.findByNombreContaining(nombre);
        return autorMapper.toDTOList(autores);
    }

    /**
     * Lista autores vivos en un año específico
     */
    public List<AutorDTO> listarVivosEnAnio(Integer anio) {
        List<Autor> autores = autorRepository.findAutoresVivosEnAnio(anio);
        return autorMapper.toDTOList(autores);
    }

    /**
     * Lista autores nacidos en un rango de años
     */
    public List<AutorDTO> listarNacidosEntre(Integer desde, Integer hasta) {
        List<Autor> autores = autorRepository.findByFechaDeNacimientoBetween(desde, hasta);
        return autorMapper.toDTOList(autores);
    }

    /**
     * Lista autores fallecidos en un rango de años
     */
    public List<AutorDTO> listarFallecidosEntre(Integer desde, Integer hasta) {
        List<Autor> autores = autorRepository.findByFechaDeFallecimientoBetween(desde, hasta);
        return autorMapper.toDTOList(autores);
    }
}
